src/Microsoft.AspNet.Server.Kestrel/Http/Connection.cs

```diff
@@ -107,7 +108,7 @@ private void ApplyConnectionFilter()
             SocketInput = filteredStreamAdapter.SocketInput;
             SocketOutput = filteredStreamAdapter.SocketOutput;
 
-            _frame = CreateFrame();
+            _frame = CreateFrame(_filterContext.ClientCertificate);
```

Comment:
It would be better if the connection filter had an abstraction that let it apply changes to the Frame directly. Ideally nothing in Microsoft.AspNet.Server.Kestrel would touch X509Certificates, they would all be managed by Microsoft.AspNet.Server.Kestrel.Https.